packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentInformation.gql.js

```diff
@@ -17,11 +17,21 @@ export const GET_PAYMENT_DETAILS = gql`
     }
 `;
 
+export const GET_PAYMENT_NONCE = gql`
```

Comment:
You should just get this from `creditCard.gql.js`

Reply:
It is not specific to CC. In general any payment method. Also, my understanding is that we were fine with duplicating GQL queries in the interest of leaving component-specific GQL queries in `component.gql.js`.

Reply:
Technically the `paymentNonce` is our own construct that we created to handle persisted payment details for display _within_ the credit card component. My point is that the parent, `paymentInformation` does not and should not be aware of what information a specific payment method needed in order to be rendered "complete".

Reply:
I agree with you totally but like you mentioned since the payment method component will be unmounted, how can we clear something from the parent which only the child has information about?

To make this simple, I said, maybe if the parent knows that all the child components if they need to save something, they will save in the `paymentNonce` key and the parent can simply delete it.

I am up for suggestions on this.

Reply:
Yea lets talk about what that prop will be named in the other thread :)